solutions/day_3/solution.py

```diff
@@ -0,0 +1,45 @@
+from typing import List
+from functools import reduce
+
+TEXT_FILE_LOCATION = "input.txt"
+LOWER_CASE_ORD_DIFFERENCE = 96
+UPPER_CASE_ORD_DIFFERENCE = 38
+
+
+def find_intersection(acc: List, val: map) -> List[int]:
+    if val != "":
+        midpoint = len(val) // 2
+        char_to_add = set(val[:midpoint]).intersection(set(val[midpoint:])).pop()
+        if char_to_add.islower():
+            acc.append(ord(char_to_add) - LOWER_CASE_ORD_DIFFERENCE)
+        else:
+            acc.append(ord(char_to_add) - UPPER_CASE_ORD_DIFFERENCE)
+    else:
+        acc += []
+    return acc
+
+
+def create_groups(acc: List[List], val: map) -> List[List]:
+    print("reached")
+    iterator = 0
+    if val != "" and iterator % 3 != 0 and iterator > 0:
+        acc.append(val)
+        iterator += 1
+    else:
+        acc += [[]]
+        acc.append(val)
+        iterator += 1
+    return acc
+
+
+def main():
+    with open(TEXT_FILE_LOCATION) as file:
+        lines = map(lambda s: s.rstrip("\n"), file.readlines())
+    elf_item_list = reduce(find_intersection, lines, [])
+    group_list = reduce(create_groups, lines, [[]])
```

Comment:
The `lines` are already consumed by the time the code reaches line 39. The output of `map` is a single-use iterator. You can only iterate through it once. In order to make it re-usable, you need to "materialize" the map iterator. That is what I do in my solutions if you notice. Whenever I need input to be iterated multiple times, I make it into a list as `list(map(lambda ..., ...))`. Then you should be good.

Reply:
Reference: https://github.com/DamianB-BitFlipper/advent-of-code-2022/blob/master/day3/solution.py#L25